### DISCUSSION TITLE
How to get random documents in a collection?

### DISCUSSION THREAD

**Discussion by alvaaz** [supabase/supabase#2304] [Questions] [answered]

**Comment by steve-chavez** [comment, marked as answer]
Similarly to the answer in https://github.com/supabase/supabase/discussions/2299#discussioncomment-986360, you need the TABLESAMPLE SQL clause to do this:

```sql
-- since you want many rows instead of one, the BERNOULLI method is more suitable
select * from <your_table> tablesample bernoulli(<number_between_1_and_100); -- the number here is the percentage of rows you want
```

Then wrap that in a SQL function and call it through [supabase rpc](https://supabase.io/docs/reference/javascript/rpc).
